docs/blog/2023-02-11-architecture.md

```diff
@@ -0,0 +1,61 @@
+---
+title: The Architecture so Far!
+description: This is the architecture of our app so far.
+slug: 2023-02-11-architecture
+authors: AbdBarho
+tags: [open-assistant]
+---
+
+I thought it would make sense to document our architecture as a blog post, since
+it would be easily visible for everyone, and it would also be nice to look back
+a couple months from now and see how stuff has changed :)
+
+A lot of work was done in the past couple of week to establish the needed
+infrastructure both for data collection and inference purposes. Thanks to
+everyone, and especially community members who supported us so far!
+
+Our current plan for the architecture of the Open Assistant is as follows:
+
+```mermaid
+flowchart TD
+  subgraph next[Web Frontend]
+    direction TB
+    website([Browser]) & nextbackend[Backend] & webpostgres[(PostgreSQL)]
+  end
+
+  subgraph data[Python backend]
+    direction TB
+    backend[Backend] & postgres[(PostgreSQL)] & redis[(Redis)]
+  end
+
+  subgraph inference[Inference Service]
+    direction TB
+    inferenceServer[Server] -.- infpostgres[(PostgreSQL)]
+  end
+
+  subgraph workers[Inference Workers]
+    direction TB
+    worker1[Worker] --- textgen1[Text\ngenerator]
+    worker2[Worker] --- textgen2[Text\ngenerator]
+    worker3[Worker] --- textgen3[Text\ngenerator]
+  end
```

Comment:
We've been playing with ActivityPub to enable multiple workers to provide feedback via `inReplyTo` and threads, mimicking human behavior. Wasn't sure where else to post so posting here. The hope is that our models can collectivly respond, and the user or users AI agent can sift through and find the responses that are most helpful to them within the context of the conversation. Something like ActivityPub based communication (Rapunzel, ATProto come to mind) would enable folks AI's to collaboratively provide their responeses.

- References
  - [WIP: IETF SCITT: Use Case: OpenSSF Metrics: activitypub extensions for security.txt](https://github.com/ietf-scitt/use-cases/blob/fd2828090482fe63a30a7ddd9e91bdb78892a01e/openssf_metrics.md#activitypub-extensions-for-securitytxt)

Cheers! 🥳